test/MorphoVaultActions.t.sol

```diff
@@ -0,0 +1,98 @@
+
+    // Contracts address on mainnet
+    address constant morphoVault = 0x8eB67A509616cd6A7c1B3c8C21D48FF57df3d458;
+    address constant USDC = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+    bytes morphoVaultActionsScripts = new YulHelper().getCode("MorphoScripts.sol/MorphoVaultActions.json");
+
+    function setUp() public {
+        // Fork setup
+        vm.createSelectFork(
+            string.concat(
+                "https://node-provider.compound.finance/ethereum-mainnet/", vm.envString("NODE_PROVIDER_BYPASS_KEY")
+            ),
+            20564787 // 2024-08-19 12:34:00 PST
+        );
+        factory = new QuarkWalletProxyFactory(address(new QuarkWallet(new CodeJar(), new QuarkStateManager())));
+    }
+
+    function testDeposit() public {
+        vm.pauseGasMetering();
+        QuarkWallet wallet = QuarkWallet(factory.create(alice, address(0)));
+
+        deal(USDC, address(wallet), 10_000e6);
+
+        QuarkWallet.QuarkOperation memory op = new QuarkOperationHelper().newBasicOpWithCalldata(
+            wallet,
+            morphoVaultActionsScripts,
+            abi.encodeWithSelector(MorphoVaultActions.deposit.selector, morphoVault, USDC, 10_000e6),
+            ScriptType.ScriptSource
+        );
+        (uint8 v, bytes32 r, bytes32 s) = new SignatureHelper().signOp(alicePrivateKey, wallet, op);
+        assertEq(IERC20(USDC).balanceOf(address(wallet)), 10_000e6);
+        assertEq(IERC4626(morphoVault).balanceOf(address(wallet)), 0);
+
+        vm.resumeGasMetering();
+        wallet.executeQuarkOperation(op, v, r, s);
+
+        assertEq(IERC20(USDC).balanceOf(address(wallet)), 0);
+        assertApproxEqAbs(IERC4626(morphoVault).balanceOf(address(wallet)), 9713.4779e18, 0.01e18);
+        assertApproxEqAbs(
+            IERC4626(morphoVault).convertToAssets(IERC4626(morphoVault).balanceOf(address(wallet))), 10_000e6, 0.01e6
+        );
+    }
+
+    function testWithdraw() public {
+        vm.pauseGasMetering();
+        QuarkWallet wallet = QuarkWallet(factory.create(alice, address(0)));
+
+        // Deal vault shares to wallet, ERC4262 is ERC20 compatible
+        deal(morphoVault, address(wallet), 10_000e18);
+
+        QuarkWallet.QuarkOperation memory op = new QuarkOperationHelper().newBasicOpWithCalldata(
+            wallet,
+            morphoVaultActionsScripts,
+            abi.encodeWithSelector(MorphoVaultActions.withdraw.selector, morphoVault, 10_000e6),
+            ScriptType.ScriptSource
+        );
+        (uint8 v, bytes32 r, bytes32 s) = new SignatureHelper().signOp(alicePrivateKey, wallet, op);
+        assertEq(IERC20(USDC).balanceOf(address(wallet)), 0e6);
+        assertEq(IERC4626(morphoVault).balanceOf(address(wallet)), 10_000e18);
+
+        vm.resumeGasMetering();
+        wallet.executeQuarkOperation(op, v, r, s);
+
+        assertEq(IERC20(USDC).balanceOf(address(wallet)), 10_000e6);
+        assertApproxEqAbs(IERC4626(morphoVault).balanceOf(address(wallet)), 286.5e18, 1e18);
```

Comment:
Again, as with other assertions regarding shares, I'm not sure how this assertion is supposed to convey information to me if I see that this test is failing.

Can we at least add a string annotation as the final call parameter? `"Expected exact shares dust in vault"` or something, perhaps.

Reply:
Yeah I agree. 
I have removed this line as rethinking a bit this doesn't mean much to assert the share value. This function only tests if `morpho` interface can handle withdraw fund correctly. So I think just asserting wallet have the withdrawn fund is good enough.